### ISSUE TITLE
Phone number format

### ISSUE THREAD

**Issue by jpbottinelli** [quarentime/quarentime-back-end#28]
I'm creating this as a bug, but it's more of a design decision that we have to make.

The current API allows any phone number format.

Based on [Twilio's](https://www.twilio.com/docs/sms/send-messages) example page  
`This parameter determines the destination phone number for your SMS message. Format this number with a '+' and a country code, e.g., +16175551212 (E.164 format).`

This is [e164 format](https://www.twilio.com/docs/glossary/what-e164). That page shows a regex that we can use and a request to validate if a number is valid (if we don't go the regex way, we need to allow some space for fake numbers for testing)

We should only allow that type of phone format and error in anything else.
For example: +5491133092321

This format also works for Whatsapp automatic phone detection and others.

**Comment by ale90bsas** [comment]
Hi, I've created a similar issue on https://github.com/quarentime/quarentime-back-end/issues/33

Please @edmarsj @mapplics help us to review this behavior as we've found that it's possible to add contacts that don't fulfill the E164 - "International Format" norm, so this becomes a blocker issue.

**Comment by ale90bsas** [comment]
Hi @mapplics, @edmarsj 

Now when I try to send an invite to an "invalid" number, I get the following error:

![image](https://user-images.githubusercontent.com/30640814/78669732-1715b880-78b3-11ea-927e-087b55b1865c.png)

So I assume that Frontend is not expecting that outcome and doesn't handle the error successfully.

However, should we check the format in Frontend before sending the request or try to "fix" it? 
Something like the described here https://developers.facebook.com/docs/whatsapp/api/contacts/

Maybe for the MVP we should just handle the error.

**Comment by mapplics** [comment]
I agree with this, maybe in the front-end we can show only contacts having the correct format, as whatsapp does. 
To be sure =>  is this correct format '+xxxxxxxxxxxx'?
@ale90bsas @edmarsj @vlatkozmejkoski @jpbottinelli

**Comment by mapplics** [comment]
@ale90bsas @edmarsj @vlatkozmejkoski @jpbottinelli in version 0.21.0, we include the regex filter proposed by @jpbottinelli Please check and let us know if is ok.

**Comment by ale90bsas** [comment]
@mapplics I've several "valid" contacts that are not shown.

For instance:

SelfTest +5491150185169
J             +5491154604534
M            +5491127281042   (I've created a new contact with the same number and it's visible)

So I'm thinking that the issue is not in the regex but in the way we got the numbers.

This is an example of a non-visible contact
![image](https://user-images.githubusercontent.com/30640814/78931250-082e3200-7a7c-11ea-8e83-7561301d9195.png)


Please let me know if I can help in the debugging of this situation.

**Comment by ale90bsas** [comment]
Hi @mapplics ,

I'm seeing more contacts than before, but I've some still missing, for instance:

SelfTest +5491150185169
J +5491154604534

Is there any additional info that I can provide here?

**Comment by ale90bsas** [comment]
Hi @mapplics,

Yesterday one of our beta testers reported the same problem when tried to add my phone (+5491150185169). We were filtering the contact...even if the number has the right format.
I instructed him to refresh the contact list, close and reopen the app but there was no way to show it.

We may want to change the approach of this validation for future versions. Perhaps show the number and if it's invalid give the user the option to manually fix it.

Best,
